templates/pkg/resource/sdk.go.tpl

```diff
@@ -164,11 +164,17 @@ func (rm *resourceManager) updateConditions (
 
 	// Terminal condition
 	var terminalCondition *ackv1alpha1.Condition = nil
+	var nonTerminalCondition *ackv1alpha1.Condition = nil
 	for _, condition := range ko.Status.Conditions {
+		// If terminal condition exists, other errors will
+		// will not be reported
```

Comment:
This isn't actually true, though, right? If a NonTerminalCondition is found first in the list of Condition structs, we will still end up "reporting" that NonTerminalCondition...

Reply:
Yes, it was an oversight and I thought we were checking for the existing TerminalCondition below, but we are actually checking if the new error is Terminal. I fixed the comment to indicate that we continue to look for TerminalCondition even if we find a RecoverableError. TerminalCondition takes higher priority to display status as the sync stops.

Reply:
Good catch
right now if someone sets terminalCondition outside of customUpdateConditions, it is reset to False in this method based on `rm.terminalAWSError` check. Should we keep the same behaviour for Recoverable as well?

Reply:
No worries, was just making sure I was reading the logic right :)